### ISSUE TITLE
Posts to Posts Importer?

### ISSUE THREAD

**Issue by markjaquith** [stuttter/wp-relationships#1]
I think a lot of people watching this project care because they're using Posts to Posts currently. What are your thoughts on the feasibility of an importer from Posts to Posts?

**Comment by JJJ** [comment]
Very feasible. Should be bundled, IMO.

- [ ] WP-CLI
- [ ] GUI
- [ ] Detect & offer-to-migrate on activation

**Comment by ejntaylor** [comment]
A few of us P2P users more likely to get involved if we can migrate their existing projects over.

I'd settle for any migration process. If the above is achievable then that would be even better.

**Comment by JJJ** [comment]
Any thoughts on how many rows y'all typically have in your P2P tables? Obviously it could be billions, but what's realistic? I've seen hundreds, but not thousands yet.

I'd like to write the MVP, and improve it from there using something like the batch processor that's already in bbPress 2.6 (GUI proven to work with millions of posts.)

**Comment by markjaquith** [comment]
> Any thoughts on how many rows y'all typically have in your P2P tables?

I'm working on a site remodel that will probably have into the low thousands. I would aim for 100k. If the schema and queries and indexes are sane, that should be easy. Most of the lookups are going to be "given some object IDs and a relationship type, show me the related things."

The order you have above is right: WP-CLI, GUI, Detect. And it's possible that, considering the audience for this, WP-CLI is good enough.

**Comment by JiveDig** [comment]
I agree with Mark. Start with CLI just to get P2P uses onboard/migrated. Add the GUI/detect later on, after the plugin is more stable and in use.

**Comment by JJJ** [comment]
Have not forgotten about this, but have not been able to prioritize it.

Trying to work ongoing development of this into a project so I can manage it via some overlap.

**Comment by lkraav** [comment]
Hi all. We're looking into many-to-many terms solutions and this popped on our radar.

Unfortunately, other than this issue discussion, it seems a bit like this idea has been abandoned. Did something else pop up in last 4 years that does many-to-many better by now, or is the base still strong here, and we would want to continue, but just resource-starved? I may be able to sponsor an amount of work to move wp-relationships forward.

@JJJ aside from an update here, we could also chat details on WP Slack, I'll DM you.
